crates/ruff/src/args.rs

```diff
@@ -806,15 +806,18 @@ impl TypedValueParser for ConfigArgumentParser {
         arg: Option<&clap::Arg>,
         value: &std::ffi::OsStr,
     ) -> Result<Self::Value, clap::Error> {
-        let path_to_config_file = PathBuf::from(value);
```

Comment:
What's the motivation for changing the ordering from path, value to value, path? Is it because `shellexpand` requires an UTF8 path? 

True, it's an edge case but I think it might be worth falling back to normal paths if it's not valid UTF 8

```
let Ok(value) = value.to_str() else {
  let path_to_config_file = PathBuf::from(value);
  if path_to_config_file.is_file() {
    return Ok(SingleConfigArgument::FilePath(path_to_config_file));
  }
  return Err(clap::error::ErrorKind::InvalidUtf8);
}
```